test/magic_number_division/main.cpp

```diff
@@ -41,6 +42,19 @@ gpu_naive_division(int32_t divisor, const int32_t* p_dividend, int32_t* p_result
     }
 }
 
+__host__ void cpu_magic_number_division(uint32_t magic_multiplier,
```

Comment:
What is this test testing?

Reply:
In develop there's no host version of magic_number_division, so any operations that involve it cannot be called on host. So I added one and a test for it.

Reply:
We don't need to call magic number division on host

Reply:
Then we should remove __host__ from all functions that uses magic division in multi_index_transform.hpp